cli/command/context/create.go

```diff
@@ -63,6 +64,7 @@ func newCreateCommand(dockerCli command.Cli) *cobra.Command {
 		"Default orchestrator for stack operations to use with this context (swarm|kubernetes|all)")
 	flags.StringToStringVar(&opts.Docker, "docker", nil, "set the docker endpoint")
 	flags.StringToStringVar(&opts.Kubernetes, "kubernetes", nil, "set the kubernetes endpoint")
+	flags.StringVar(&opts.From, "from", "", "create context from a named context")
```

Comment:
The default depends on whether you use `--docker`/`--kubernetes` or not. With then it will use the options given to those arguments, without it will create from the current context. Seems a bit finicky to me I'll admit.
To make it less so we could remove the defaulting entirely but then creating from the current/environment is a bit more of a pain.